packages/wasm/crate/src/planner.rs

```diff
@@ -1,591 +1,389 @@
+use crate::utils;
+use crate::{error::WasmResult, swap_record::SwapRecord};
+use anyhow::anyhow;
+use ark_ff::UniformRand;
+use decaf377::Fq;
+use penumbra_asset::{asset, Balance, Value};
+use penumbra_dex::swap_claim::SwapClaimPlan;
 use penumbra_dex::{
-    lp::action::{PositionClose, PositionOpen},
-    lp::plan::PositionWithdrawPlan,
-    lp::position::{self, Position},
-    lp::Reserves,
-    swap::SwapPlaintext,
-    swap::SwapPlan,
-    swap_claim::SwapClaimPlan,
+    swap::{SwapPlaintext, SwapPlan},
     TradingPair,
 };
-use penumbra_fee::{Fee, FeeTier, GasPrices};
-use penumbra_governance::{
-    proposal_state::Outcome as ProposalOutcome, DelegatorVotePlan, Proposal, ProposalDepositClaim,
-    ProposalSubmit, ProposalWithdraw, ValidatorVote, Vote,
-};
-use penumbra_ibc::IbcRelay;
+use penumbra_fee::{Fee, FeeTier, Gas, GasPrices};
+use penumbra_keys::keys::AddressIndex;
 use penumbra_keys::Address;
 use penumbra_num::Amount;
-use penumbra_proto::core::keys::v1::AddressIndex;
-use penumbra_proto::view::v1::{NotesForVotingRequest, NotesRequest};
-use penumbra_shielded_pool::{fmd, Ics20Withdrawal, Note, OutputPlan, SpendPlan};
-use penumbra_stake::{rate::RateData, validator};
-use penumbra_stake::{IdentityKey, UndelegateClaimPlan};
-use penumbra_tct as tct;
-use penumbra_transaction::gas::{self, GasCost};
-use penumbra_transaction::{
-    memo::MemoPlaintext,
-    plan::{ActionPlan, MemoPlan, TransactionPlan},
+use penumbra_sct::params::SctParameters;
+use penumbra_shielded_pool::{fmd, OutputPlan, SpendPlan};
+use penumbra_stake::rate::RateData;
+use penumbra_transaction::gas::GasCost;
+use penumbra_transaction::memo::MemoPlaintext;
+use rand_core::OsRng;
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen::JsValue;
+
+use penumbra_proto::view::v1::{
+    transaction_planner_request as tpr, NotesRequest, TransactionPlannerRequest,
 };
+use penumbra_transaction::{plan::MemoPlan, ActionPlan, TransactionParameters, TransactionPlan};
 
-use crate::note_record::SpendableNoteRecord;
-
-/// A planner for a [`TransactionPlan`] that can fill in the required spends and change outputs upon
-/// finalization to make a transaction balance.
-pub struct Planner<R: RngCore + CryptoRng> {
-    rng: R,
-    balance: Balance,
-    vote_intents: BTreeMap<u64, VoteIntent>,
-    plan: TransactionPlan,
-    ibc_actions: Vec<IbcRelay>,
-    gas_prices: GasPrices,
-    fee_tier: FeeTier,
-    // IMPORTANT: if you add more fields here, make sure to clear them when the planner is finished
-}
-
-#[derive(Debug, Clone)]
-struct VoteIntent {
-    start_block_height: u64,
-    start_position: tct::Position,
-    rate_data: BTreeMap<IdentityKey, RateData>,
-    vote: Vote,
-}
-
-impl<R: RngCore + CryptoRng> Debug for Planner<R> {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        f.debug_struct("Builder")
-            .field("balance", &self.balance)
-            .field("plan", &self.plan)
-            .finish()
-    }
+struct ActionList {
```

Comment:
This was intended as just an action list but grew to become something like a very very barebones planner-backbone. However, I think it's probably good to keep decoupled from the Rust view server implementation, considering how we ended up here.